### ISSUE TITLE
The `flip` method doesn't reverse the argument's order of a partially applied function

### ISSUE THREAD

**Issue by sebastienfilion** [slavaGanzin/ramda.py#4]
The `flip` method doesn't reverse the argument's order of a partially applied function. The following example demonstrates the issue; both `add` and `multiply` are partially applied `reduce`. The functions are still expecting an accumulator and a list of values.  
Flipping the partially applied `reduce` should expect a list of values and an accumulator.  
Currently, this doesn't happen which results in an error `TypeError: 'int' object is not iterable`.

```python
import ramda as R

# add :: Number → [Number] → Number 
add = R.reduce(lambda a, b: a + b)

# multiply :: Number → [Number] → Number 
multiply = R.reduce(lambda a, b: a * b)

total = R.compose(
    R.flip(multiply)([1, 2, 3]),
    R.flip(add)([1, 2, 3])
)(0)

assert total == 36
```

Using the Javascript library, it works as expected.

```javascript
const assert = require('assert');
const R = require('ramda');

// add :: Number → [Number] → Number 
const add = R.reduce((a, b) => a + b);

// multiply :: Number → [Number] → Number 
const multiply = R.reduce((a, b) => a * b);

const total = R.compose(
  R.flip(multiply)([1, 2, 3]),
  R.flip(add)([1, 2, 3])
)(0);

assert.strictEqual(total, 36);
```

**Comment by slavaGanzin** [comment]
@sebastienfilion Thanks for  your exceptional report. 

fix published as 0.5.7

**Comment by sebastienfilion** [comment]
Thank you for your excellent work.
